src/plugins/intel_cpu/src/utils/clone_original_blob.cpp

```diff
@@ -0,0 +1,120 @@
+        }
+    } else {
+        if (prec == element::string) {
+            memory = std::make_shared<StringMemory>(engine, memDesc);
+            auto src = blob.getDataAs<StringMemory::OvString>();
+            auto dst = memory->getDataAs<StringMemory::OvString>();
+            std::copy(src, src + size, dst);
+        } else {
+            memory = std::make_shared<Memory>(engine, memDesc);
+            memcpy(memory->getData(), blob.getData(), blob.getSize());
+        }
+    }
+
+    MemoryPtr ptr;
+    if (memDesc.getPrecision() == element::string) {
+        ptr = std::make_shared<StringMemory>(engine, memDesc);
+    } else {
+        ptr = std::make_shared<StaticMemory>(engine, memDesc);
+    }
+
+    ptr->load(*memory.get(), needFlushDenormalsToZero);
+
+    return ptr;
+}
+
+InputPrepType requiresPreProcessing(const IMemory& blob, GraphContext::CPtr context, const dnnl::engine& engine) {
+    const auto shape = blob.getShape();
+    const auto prec = blob.getPrecision();
+
+    // DAZ has been set, processor automatically converts all denormal source operands
+    // to a zero with the sign of the original operand before performing any
+    // computations on them, thus no need to flush them to zero manually
+    bool needFlushDenormalsToZero = context->getConfig().DAZOn ? false : true;
+
+    auto isBlobAligned = [&] () {
+        bool blobAlignedOnSSE = true;
+#if defined(OPENVINO_ARCH_X86) || defined(OPENVINO_ARCH_X86_64)
+        // Majority of arithmetic and data processing instructions in legacy SSE isa requires
+        // the memory address in the operands must be aligned on 16-byte boundary. To ensure
+        // safely reusing ngraph const blob memory, need to check address alignment.
+        const void *ptr = blob.getData();
+        blobAlignedOnSSE = mayiuse(dnnl::impl::cpu::x64::avx2) || ((reinterpret_cast<uintptr_t>(ptr) & 15) == 0);
+#endif
+        return blobAlignedOnSSE;
+    };
+
+    // @WARNING The order of the checks below matters
+    // The checks are ordered from lightweight to heavy
+    if (prec == element::string) {
+        DEBUG_LOG("Clone is necessary for a string Constant");
+        return InputPrepType::SimpleClone;
+    }
+
+    const bool mustFlushDenormalsToZero = needFlushDenormalsToZero && std::make_shared<HasSubnormals>()->execute(blob);
+    if (mustFlushDenormalsToZero) {
+        DEBUG_LOG("Clone is necessary for Constant containing subnormals");
+        return InputPrepType::FTZ;
+    }
+
+    if (!isBlobAligned()) {
+        DEBUG_LOG("Clone is necessary for not aligned blobs");
+        return InputPrepType::SimpleClone;
+    }
+
+    if (context->getWeightsCache() &&
+        context->getNumNumaNodes() > 1 &&
+        context->getCPUStreamExecutor()->get_streams_num() > 1) {
+        DEBUG_LOG("Clone is necessary for multistream multisocket configuration");
+        return InputPrepType::PutToNumaLocalCache;
+    }
```

Comment:
Maybe I misunderstood the question, but the idea is a bit different.
If we need to flush subnormals to zero based on provided configuration, that means we have to always check whether subnormals really exist in a Tensor. We must do it unconditionally. So, there is no need to perform the check below, if we are going to clone the tensor because of subnormals anyway.